agent/bench-scripts/pbench-fio

```diff
@@ -496,46 +473,62 @@ function record_iteration {
 	echo ${iteration} | pbench-add-metalog-option ${mdlog} iterations/${iteration} iteration_name
 }
 
-# Ensure the right version of the benchmark is installed
-function fio_install() {
-	check_install_rpm "${benchmark_rpm}" "${ver}" "${match}"
-	if [ ! -z "${clients}" ] ; then
-		debug_log "verifying clients have fio installed"
-		echo "verifying clients have fio installed"
-		for client in "${client_names[@]}" ; do
-			ssh $ssh_opts $client ${pbench_install_dir}/bench-scripts/$script_name --install &
-		done
-		# FIXME - save the pid and wait for each pid to ensure it returns with success
-		wait
+function local_pre_check() {
+	local devs="${1}"
+	local ver="${2}"
+	local match="${3}"
+
+	debug_log "local_pre_check(devs=\"${devs}\", ver=\"${ver}\", match=\"${match}\")"
+
+	# First check for the supported version of fio in use.
+
+	local _stdout=$(${benchmark_bin} --version)
+	local ver_installed=${_stdout##*-}
+
+	vercmp "${benchmark}" "${match}" "${ver}" "${ver_installed}"
+	if [[ ${?} -ne 0 ]]; then
+		return 1
 	fi
-}
 
-# Make sure this devices exists
-function fio_device_check() {
-	local devs=$1
-	local clients=$2
+	# Now if we have any block devices specified, verify them.
 
 	local dev=""
-	local client=""
 	local rc=0
-
-	debug_log "fio_device_check(devs=\"${devs}\", clients=\"${clients}\")"
 	for dev in ${devs//,/ }; do
-		if echo $dev | grep -q "^/dev/"; then
-			if [ ! -z "${clients}" ]; then
-				for client in "${client_names[@]}" ; do
-					debug_log "checking to see if $dev exists on client $client"
-					ssh $ssh_opts $client "if [ -L $dev ]; then dev=`dirname $dev`/`readlink $dev`; fi; test -b $dev" || rc=1
-				done
-			else
-				if [ -L $dev ]; then dev=`dirname $dev`/`readlink $dev`; fi; test -b $dev || rc=1
-			fi
-			if [ $rc -eq 1 ]; then
-				error_log "At least one client did not have block device $dev, exiting"
-				exit 1
+		if [[ ${dev} =~ ^/dev/ ]]; then
+			local rp_dev=$(realpath ${dev})
+			if [[ ! -b ${rp_dev} ]]; then
+				error_log "${pbench_hostname_full} does not have block device ${dev}"
 			fi
 		fi
 	done
+
+	return ${rc}
+}
+
+function fio_pre_check() {
+	local targets="${1}"
+	local clients="${2}"
+	local ver="${3}"
+	local match="${4}"
+
+	local rc=0
+
+	debug_log "fio_pre_check(targets=\"${targets}\", clients=\"${clients}\", ver=\"${ver}\", match=\"${match}\")"
+	if [ ! -z "${clients}" ]; then
+		local client=""
+		for client in "${client_names[@]}" ; do
+			debug_log "Running pre-check on client ${client}"
+			ssh ${ssh_opts} ${client} pbench-fio --pre-check --targets="${targets}" "${ver}" "${match}" || rc=1
+		done
+	else
+		local_pre_check "${devs}" "${ver}" "${match}"
+		rc=${?}
+	fi
```

Comment:
In what I believe is the analogous code in `pbench-uperf`, the local and remote tests are decoupled, so that the script might execute _both_ of them, rather than either-or.

Reply:
I can add a commit, or a separate PR to endow `pbench-fio` with a better understanding of "local host", shared with `pbench-perf`.  But that is a big more invasive as `pbench-uperf` had some semblance of an understanding of local vs remote, where as `pbench-fio` does not at all right now.

Reply:
This is being pursued in #2542.